components/dashboard/src/settings/Integrations.tsx

```diff
@@ -477,7 +477,16 @@ export function GitIntegrationModal(props: ({
     useEffect(() => {
         setErrorMessage(undefined);
         validate();
-    }, [clientId, clientSecret])
+    }, [clientId, clientSecret, type])
+
+    useEffect(() => {
+        if (props.mode === "new") {
+            // If the host value has been modified e.g. not gitlab.example.com, assume it has been set by user and end operation
+            if (!host.includes(".example.com")) return;
+            const exampleHostname = `${type.toLowerCase()}.example.com`;
+            updateHostValue(exampleHostname);
+        }
+    }, [type]);
```

Comment:
This hook is for a little UX improvement to modify `host` and `redirectURL` whenever the provider changes, that being said, I understand the team is following the Minimal Valuable Change approach and perhaps this creeps a bit beyond the scope of the issue I set out to address. If that's the case I am happy to remove the hook.